File: ajps-backend/src/main/java/com/himusharier/ajps_backend/model/Issue.java
package com.himusharier.ajps_backend.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.himusharier.ajps_backend.constants.IssueStatus;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "ajps_issues")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Issue {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private int volume;

    @Column(nullable = false)
    private int number;

    @Column(nullable = false)
    private LocalDate publicationDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private IssueStatus status;

    // Many-to-One relationship with Journal
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "journal_id", nullable = false)
    @JsonBackReference // To prevent infinite recursion in JSON serialization
    private Journal journal;

    @OneToMany(mappedBy = "issue", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private List<Paper> papers = new ArrayList<>();

//    @ManyToOne
//    @JoinColumn(name = "submission_id")
//    private Submission submission;
}

File: ajps_flutter_app/lib/models/issue.dart
import 'package:ajps_flutter_app/models/paper.dart';

class Issue {
  final int id;
  final int volume;
  final int number;
  final DateTime publicationDate;
  final String status;
  final List<Paper> papers;

  Issue({
    required this.id,
    required this.volume,
    required this.number,
    required this.publicationDate,
    required this.status,
    required this.papers,
  });

  factory Issue.fromJson(Map<String, dynamic> json) {
    return Issue(
      id: json['id'],
      volume: json['volume'],
      number: json['number'],
      publicationDate: DateTime.parse(json['publicationDate']),
      status: json['status'],
      papers: (json['papers'] as List).map((p) => Paper.fromJson(p)).toList(),
    );
  }
}
